windlass/pins.py

```diff
@@ -400,6 +412,7 @@ def read_pins(self, repodir=None):
                             actual_filename=filename,
                         )
                     )
+            pin_stream.close()
```

Comment:
Would it not possibly conflict somehow with "with" inside yielding functions?

Reply:
good point, I think that closing the same file twice doesn't trigger an error, so can either use explicit and remove the `with` from the iter_pin_files? or maybe it would be more correct that the file is returned there and opened here as its not called iter_pin_file_streams?

Reply:
I think it would be best to drop it from here, I was not aware you could mix "with" and "yield" at time I worked on this.